server/src/main/java/org/opensearch/search/sandboxing/cancellation/DefaultTaskCancellation.java

```diff
@@ -0,0 +1,79 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.search.sandboxing.cancellation;
+
+import org.opensearch.cluster.metadata.Sandbox;
+import org.opensearch.search.sandboxing.SandboxLevelResourceUsageView;
+import org.opensearch.search.sandboxing.resourcetype.SandboxResourceType;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Represents the default task cancellation strategy.
+ * This class extends the AbstractTaskCancellation and provides a method to get the sandboxes from which tasks should be cancelled.
+ */
+public class DefaultTaskCancellation extends AbstractTaskCancellation {
+    public DefaultTaskCancellation(
+        TaskSelectionStrategy cancellationStrategy,
+        Map<String, SandboxLevelResourceUsageView> sandboxLevelViews,
+        Set<Sandbox> activeSandboxes
+    ) {
+        super(cancellationStrategy, sandboxLevelViews, activeSandboxes);
+    }
+
+    /**
+     * // TODO create another Task cancellation implementation that does below
+     *     - if node not in duress
+     *         - pick sandboxes in enforced mode only
+     *     - if node in duress
+     *         - pick sandboxes in enforced mode
+     *         - tasks running in deleted sandboxes with tasks running
+     *         - pick sandboxes in enforced mode
+     */
+    public List<Sandbox> getSandboxesToCancelFrom() {
+        final List<Sandbox> sandboxesToCancelFrom = new ArrayList<>();
+
+        for (Sandbox sandbox : this.activeSandboxes) {
+            Map<SandboxResourceType, Long> currentResourceUsage = getResourceUsage(sandbox.getId());
+
+            for (Sandbox.ResourceLimit resourceLimit : sandbox.getResourceLimits()) {
+                if (isBreachingThreshold(currentResourceUsage, resourceLimit)) {
+                    sandboxesToCancelFrom.add(sandbox);
+                    break;
+                }
+            }
+        }
+
+        return sandboxesToCancelFrom;
+    }
+
+    /**
+     * Checks if the current resource usage is breaching the threshold of the provided resource limit.
+     *
+     * @param currentResourceUsage The current resource usage
+     * @param resourceLimit The resource limit to check against
+     * @return true if the current resource usage is breaching the threshold, false otherwise
+     */
+    private boolean isBreachingThreshold(Map<SandboxResourceType, Long> currentResourceUsage, Sandbox.ResourceLimit resourceLimit) {
+        return currentResourceUsage.get(resourceLimit.getResourceType()) > resourceLimit.getThreshold();
+    }
+
+    /**
+     * Returns the resource usage of the sandbox with the provided ID.
+     *
+     * @param sandboxId The ID of the sandbox
+     * @return The resource usage of the sandbox
+     */
+    private Map<SandboxResourceType, Long> getResourceUsage(String sandboxId) {
+        return sandboxLevelViews.get(sandboxId).getResourceUsageData();
```

Comment:
I dont think so, here's my thinking correct me if i'm wrong 
SandboxService on its every iteration does this (in order)
- group all search tasks and their resource usages into sandboxLevelResourceUsageViews.
- activeSandboxes list is derived from the cluster state.

in between the 2 steps, a sandbox can created / deleted. 
if a sandbox is created, 
	a task could have been created and tagged with this sandbox, if it does not get picked up in the first step, it will be picked up in the next iteration. However, the activeSandboxes can include this sandbox and hence it would fail in the step of `sandboxLevelViews.get(sandboxId)` since sandboxId is picked from the activeSandboxes list and will might not be present in sandboxLevelResourceUsageViews. 
	
Good catch, let me address this.